### PR TITLE
Fixes for `gdb` stubs

### PR DESCRIPTION
**Pull request by LukeShu**
I've noticed several mistakes in the `gdb` stubs; I've fixed each in a separate commit. See the individual commit messages for details.

### CODE REVIEW

**Comment by LukeShu** [comment]
> Diff from [mypy_primer](https://github.com/hauntsaninja/mypy_primer), showing the effect of this PR on open source code:
> 
> ```diff
> pwndbg (https://github.com/pwndbg/pwndbg)
> + pwndbg/aglib/disasm/__init__.py:34: note: In module imported here:
> ```

I'm not sure what this means or what it has to do with my changes.

File: stubs/gdb/gdb/__init__.pyi [hunk 1]
```diff
@@ -663,18 +669,142 @@ class LineTable:
 
 # Breakpoints
 
+# The list of `case`s in the big `switch` block in py-breakpoint.c:bppy_init().
+_BreakpointType_NonWatchpoint: TypeAlias = Literal[BP_BREAKPOINT, BP_HARDWARE_BREAKPOINT, BP_CATCHPOINT]
+_BreakpointType_Watchpoint: TypeAlias = Literal[BP_WATCHPOINT]
+
+_WatchpointClass: TypeAlias = Literal[WP_READ, WP_WRITE, WP_ACCESS]
```
Comment (LukeShu) [inline stubs/gdb/gdb/__init__.pyi:676]: Huh, mypy seems happy enough with this when checking my code that uses these hints, but in and pyright are upset about it when checking the stubs themselves.

> Variable not allowed in type expression (reportInvalidTypeForm)
> "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value (reportInvalidTypeForm)

Is there a proper way to write this? Should I turn `BP_*` and `WP_*` into enums? (They're enums in C, but the Python bindings have them as plain `int`s, not `enum.Enum`s).

**Comment by brianschubert** [comment]
> I'm not sure what this means or what it has to do with my changes.

The referenced line is a `gdb` import. There are errors that would usually follow, but mypy primer is filtering them out since they're inside typeshed stubs. Take a look at the failing jobs for what needs to be fixed.

File: stubs/gdb/gdb/__init__.pyi (hunk 1 above)
Comment (brianschubert) [reply stubs/gdb/gdb/__init__.pyi]: Are the integer values stable? If so, you could probably fill in the appropriate values in both the constants and the literal alias.

**Comment by LukeShu** [comment]
OK, I've
- fixed a mis-copied `@overload`
- given up on typing the enums beyond `int`, unless someone can tell me the proper way to do it

File: stubs/gdb/gdb/__init__.pyi (hunk 1 above)
Comment (LukeShu) [reply stubs/gdb/gdb/__init__.pyi]: I don't believe they're stable.

File: stubs/gdb/gdb/__init__.pyi [hunk 2]
```diff
@@ -289,7 +288,14 @@ class PendingFrame:
 class UnwindInfo:
     def add_saved_register(self, reg: str | RegisterDescriptor | int, value: Value, /) -> None: ...
 
-frame_unwinders: list[gdb.unwinder.Unwinder]
+class _Unwinder(Protocol):
```
Comment (srittau) [inline stubs/gdb/gdb/__init__.pyi:291]: I assume this doesn't exist at runtime. Could you add a `@type_check_only` decorator (needs import from `typing`)?

```suggestion
@type_check_only
class _Unwinder(Protocol):
```

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks! One minor request.

File: stubs/gdb/gdb/__init__.pyi (hunk 2 above)
Comment (LukeShu) [reply stubs/gdb/gdb/__init__.pyi:292]: Done.

**Review by srittau — APPROVED** [review]
For future reference: Please don't force push branches as this makes it harder for us to review changes. We squash on merge anyway. Thanks again!

**Pull request merged**